File: app/src/main/java/de/cineaste/android/database/UserDbHelper.java
package de.cineaste.android.database;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;

import de.cineaste.android.entity.User;

public class UserDbHelper extends BaseDao {

	private static UserDbHelper mInstance;

	private UserDbHelper(Context context) {
		super(context);
	}

	public static UserDbHelper getInstance(Context context) {
		if (mInstance == null) {
			mInstance = new UserDbHelper(context);
		}

		return mInstance;
	}


	public void createUser(User user) {
		ContentValues values = new ContentValues();
		values.put(UserEntry.COLUMN_USER_NAME, user.getUserName());

		writeDb.insert(UserEntry.TABLE_NAME, null, values);
	}

	public User getUser() {

		String[] projection = {
				UserEntry._ID,
				UserEntry.COLUMN_USER_NAME
		};

		Cursor c = readDb.query(
				UserEntry.TABLE_NAME,
				projection,
				null,
				null,
				null,
				null,
				null,
				null);

		User user = null;

		if (c.moveToFirst()) {
			do {
				user = new User();
				user.setUserName(
						c.getString(c.getColumnIndexOrThrow(UserEntry.COLUMN_USER_NAME)));
			} while (c.moveToNext());
		}
		c.close();
		return user;
	}
}

File: app/src/main/kotlin/de/cineaste/android/database/dbHelper/UserDbHelper.kt
package de.cineaste.android.database.dbHelper

import android.content.ContentValues
import android.content.Context
import de.cineaste.android.database.dao.BaseDao
import de.cineaste.android.entity.User

class UserDbHelper private constructor(context: Context) : BaseDao(context) {

    val user: User?
        get() {

            val projection = arrayOf(UserEntry.ID, UserEntry.COLUMN_USER_NAME)

            val c = readDb.query(
                UserEntry.TABLE_NAME,
                projection,
                null, null, null, null, null, null
            )

            var user: User? = null

            if (c.moveToFirst()) {
                do {
                    user = User()
                    user.userName = c.getString(c.getColumnIndexOrThrow(UserEntry.COLUMN_USER_NAME))
                } while (c.moveToNext())
            }
            c.close()
            return user
        }

    fun createUser(user: User) {
        val values = ContentValues()
        values.put(UserEntry.COLUMN_USER_NAME, user.userName)

        writeDb.insert(UserEntry.TABLE_NAME, null, values)
    }

    companion object {

        private var mInstance: UserDbHelper? = null

        fun getInstance(context: Context): UserDbHelper {
            if (mInstance == null) {
                mInstance = UserDbHelper(context)
            }

            return mInstance!!
        }
    }
}